### PR TITLE
Add tchannel option to agent builder

### PR DESCRIPTION
**Pull request by black-adder**

### CODE REVIEW

**Comment by yurishkuro** [comment]
lgtm, but can you explain the use case?

**Comment by black-adder** [comment]
github/tchannel-go doesn't know about muttley. We need to use gocommon/xtchannel internally which can be configured to use muttley. Adding this option to allow this.

File: cmd/agent/app/agent_test.go [hunk 1]
```diff
@@ -57,6 +57,7 @@ func TestAgentStartStop(t *testing.T) {
 	require.NoError(t, err)
 	ch := make(chan error, 2)
 	go func() {
+		time.Sleep(time.Second * 2)
```
Comment (yurishkuro) [inline cmd/agent/app/agent_test.go]: why do we need to delay start up of the agent?

Comment (black-adder) [inline cmd/agent/app/agent_test.go]: oops, sorry forgot to remove

File: cmd/agent/app/config_test.go [hunk 2]
```diff
@@ -28,12 +28,14 @@ import (
...
-	"go.uber.org/zap"
-	"gopkg.in/yaml.v2"
+	"github.com/uber/tchannel-go"
 
 	"github.com/uber/jaeger/pkg/discovery"
 	"github.com/uber/jaeger/thrift-gen/jaeger"
 	"github.com/uber/jaeger/thrift-gen/zipkincore"
+
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v2"
```
Comment (yurishkuro) [inline cmd/agent/app/config_test.go]: up

**Pull request merged**